pyteal/ast/abi/tuple_test.py

```diff
@@ -824,3 +824,67 @@ def test_TupleElement_store_into():
 
             with pt.TealComponent.Context.ignoreExprEquality():
                 assert actual == expected, "Test at index {} failed".format(i)
+
+
+class NT_0(abi.NamedTuple):
+    f0: abi.Uint64
+    f1: abi.Uint32
+    f2: abi.Uint16
+    f3: abi.Uint8
+
+
+class NT_1(abi.NamedTuple):
+    f0: abi.StaticArray[abi.Bool, Literal[4]]
+    f1: abi.DynamicArray[abi.String]
+    f2: abi.String
+    f3: abi.Bool
+    f4: abi.Address
+    f5: NT_0
+
+
+class NT_2(abi.NamedTuple):
+    f0: abi.Bool
+    f1: abi.Bool
+    f2: abi.Bool
+    f3: abi.Bool
+    f4: abi.Bool
+    f5: abi.Bool
+    f6: abi.Bool
+    f7: abi.Bool
+    f8: NT_1
+
+
+class NT_3(abi.NamedTuple):
+    f0: NT_0
+    f1: NT_1
+    f2: NT_2
+
+
+@pytest.mark.parametrize("test_case", [NT_0, NT_1, NT_2, NT_3])
+def test_NamedTuple_getitem(test_case: type[abi.NamedTuple]):
+    tuple_value = test_case()
+    tuple_len_static = tuple_value.type_spec().length_static()
+    for i in range(tuple_len_static):
+        elem_by_field: abi.TupleElement = getattr(tuple_value, f"f{i}")
+        assert (
+            type(elem_by_field) is abi.TupleElement
+        ), f"Test case {test_case} at field f{i} must be TupleElement"
+        elem_by_index: abi.TupleElement = tuple_value[i]
+        assert (
+            type(elem_by_field) is abi.TupleElement
+        ), f"Test case {test_case} at index {i} must be TupleElement"
+        assert (
+            elem_by_field.index == i
```

Comment:
I'd like to see the `index` and `tuple` properties of `elem_by_index` tested as well, to ensure the `__getitem__` function behaves exactly the same on `NamedTuple`